libnetwork/osl/interface_linux.go

```diff
@@ -499,6 +538,29 @@ func (n *Namespace) prepAdvertiseAddrs(ctx context.Context, i *Interface, ifInde
 	return ua, un
 }
 
+// ValidateAdvertiseAddrNMsgs returns an error if count is not within the allowed
+// range for the number of unsolicited ARP/NA messages sent to advertise an
+// interface's addresses when it is configured.
+func ValidateAdvertiseAddrNMsgs(count int) error {
```

Comment:
The `osl` pkg isn't interested in validating user input. Could you move both funcs into `libnetwork` please?

Reply:
The user-input validation [is in libnetwork](https://github.com/robmry/moby/blob/68a7767ae0ef9b20d7e8dcccca5221a989d7a7b6/libnetwork/network.go#L546-L559) - but, once it's got as far as having an integer, it asks `osl` to range-check it. The `osl` package sends the ARP/NA messages, so I think it makes sense for the rules about what it's willing to send to live there too?

Reply:
I've done that - and added range checks to `osl`'s option setters.